synapse/handlers/profile.py

```diff
@@ -86,9 +89,12 @@ def set_displayname(self, target_user, requester, new_displayname):
     @defer.inlineCallbacks
     def get_avatar_url(self, target_user):
         if self.hs.is_mine(target_user):
-            avatar_url = yield self.store.get_profile_avatar_url(
-                target_user.localpart
-            )
+            try:
+                avatar_url = yield self.store.get_profile_avatar_url(
```

Comment:
It probably makes more sense to make the storage function return `None` rather than exposing `StoreError`. The `_simple_select_one*` functions should take a `allow_none` param.